### PR TITLE
enable beam_search path for text-generation

### LINKED ISSUE #270: Enable beam_search for text-generation
### Feature request

currently test-generation only support beam=1,which is not exposed to users, any possible that expose --beams user?

### Motivation

enable beam_search algo path if beams > 1 and beams =1 falls into greedy_search

### Your contribution

submit a PR

### PR DESCRIPTION
**Pull request by ZhaiFeiyue**
# What does this PR do?
as title

Fixes  #270 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: examples/text-generation/run_generation.py [hunk 1]
```diff
@@ -80,6 +80,12 @@ def main():
         action="store_true",
         help="Whether to use sampling for generation.",
     )
+    parser.add_argument(
+        "--beams",
+        default=1,
+        type=int,
+        help="Number of beams used by beam_search algo for generation.",
```
Comment (regisss) [inline examples/text-generation/run_generation.py:87]:
```suggestion
        "--num_beams",
        default=1,
        type=int,
        help="Number of beams used for beam search generation. 1 means greedy search will be performed.",
```

File: examples/text-generation/run_generation.py [hunk 2]
```diff
@@ -219,6 +225,7 @@ def main():
         max_new_tokens=args.max_new_tokens,
         use_cache=args.use_kv_cache,
         do_sample=args.do_sample,
+        num_beams=args.beams,
```
Comment (regisss) [inline examples/text-generation/run_generation.py:228]:
```suggestion
        num_beams=args.num_beams,
```

**Review by regisss — COMMENTED** [review]
Nice to enable beam search with static shapes @ZhaiFeiyue :rocket: 

I just left a couple of comments to change the arg name to `num_beams` so that we stay consistent with Transformers.

File: examples/text-generation/run_generation.py (hunk 1 above)
Comment (ZhaiFeiyue) [reply examples/text-generation/run_generation.py]: done

**Review by regisss — APPROVED** [review]
Failing tests are not related to this PR so it's good to merge :+1:

**Pull request merged**